benchalerts/benchalerts/conbench_dataclasses.py

```diff
@@ -0,0 +1,254 @@
+"""Dataclasses for organizing and storing relevant info from Conbench."""
+
+from dataclasses import dataclass
+from typing import List, Optional
+
+from benchclients.logging import fatal_and_log
+
+
+@dataclass
+class BenchmarkResultInfo:
+    """Track and organize specific info about one BenchmarkResult."""
+
+    name: str
+    link: str
+    has_error: bool
+    has_z_regression: bool
+    run_id: str
+    run_reason: str
+    run_time: str
+    run_link: str
+
+
+@dataclass
+class RunComparisonInfo:
+    """Track and organize specific info about a comparison between a contender run and
+    its baseline run.
+
+    Parameters
+    ----------
+    contender_info
+        The dict returned from Conbench when hitting /runs/{contender_run_id}. Contains
+        info about the run's ID, commit, errors, links, etc.
+    baseline_info
+        The dict returned from Conbench when hitting /runs/{baseline_run_id}, if a
```

Comment:
Once again strong motivator for introducing proper types for these objects, so that in my IDE can make use of rich type information.